docs: deprecate SHA notice [DET-4336] (#1500)
determined-ai/determined: docs/topic-guides/hp-tuning-det/index.txt

Synchronous Searchers
=====================

> Note: Adaptive (Simple) and Adaptive (Advanced) to be deprecated.

Reviewer: ```suggestion
Note: Adaptive (Simple) and Adaptive (Advanced) will be deprecated.
```

> Note: Adaptive (Simple) and Adaptive (Advanced ) to be deprecated.

Reviewer 2: Whitespace.

> Recommended to use state-of-the-art Adaptive (ASHA).

Reviewer: ```suggestion
We recommend using the state-of-the-art Adaptive (ASHA) searcher.
```
